src/components/switch/index.tsx

```diff
@@ -1,87 +1,86 @@
 
+export type SwitchProps = {
+  /**
+   * The value of the switch. If true the switch will be turned on. Default value is false
+   */
+  value?: boolean;
+  /**
+   * Invoked with the new value when the value changes
+   */
+  onValueChange?: (value: boolean) => void;
+  /**
+   * Whether the switch should be disabled
+   */
+  disabled?: boolean;
+  /**
+   * The Switch width
+   */
+  width?: number;
+  /**
+   * The Switch height
+   */
+  height?: number;
+  /**
+   * The Switch background color when it's turned on
+   */
+  onColor?: string;
+  /**
+   * The Switch background color when it's turned off
+   */
+  offColor?: string;
+  /**
+   * The Switch background color when it's disabled
+   */
+  disabledColor?: string;
+  /**
+   * The Switch's thumb color
+   */
+  thumbColor?: string;
+  /**
+   * The Switch's thumb size (width & height)
+   */
+  thumbSize?: number;
+  /**
+   * The Switch's thumb style
+   */
+  thumbStyle?: object | number | []; 
+  style?: any;
+  testID?: string; 
+}
+
 /**
  * Switch component for toggling boolean value related to some context
  */
-class Switch extends BaseComponent {
+class Switch extends Component<SwitchProps> {
   static displayName = 'Switch';
-
-  static propTypes = {
-    /**
-     * The value of the switch. If true the switch will be turned on. Default value is false
-     */
-    value: PropTypes.bool,
-    /**
-     * Invoked with the new value when the value changes
-     */
-    onValueChange: PropTypes.func,
-    /**
-     * Whether the switch should be disabled
-     */
-    disabled: PropTypes.bool,
-    /**
-     * The Switch width
-     */
-    width: PropTypes.number,
-    /**
-     * The Switch height
-     */
-    height: PropTypes.number,
-    /**
-     * The Switch background color when it's turned on
-     */
-    onColor: PropTypes.string,
-    /**
-     * The Switch background color when it's turned off
-     */
-    offColor: PropTypes.string,
-    /**
-     * The Switch background color when it's disabled
-     */
-    disabledColor: PropTypes.string,
-    /**
-     * The Switch's thumb color
-     */
-    thumbColor: PropTypes.string,
-    /**
-     * The Switch's thumb size (width & height)
-     */
-    thumbSize: PropTypes.number,
-    /**
-     * The Switch's thumb style
-     */
-    thumbStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number, PropTypes.array])
-  };
-
+  
   state = {
     thumbPosition: new Animated.Value(this.props.value ? 1 : 0)
   };
 
-  generateStyles() {
-    this.styles = createStyles(this.getThemeProps());
-  }
-
-  componentDidUpdate(prevProps) {
-    const {value} = this.getThemeProps();
+  styles = createStyles(this.props);
+  
+  componentDidUpdate(prevProps: SwitchProps) {
+    const {value} = this.props;
     if (prevProps.value !== value) {
       this.toggle(value);
     }
   }
 
   getAccessibilityProps() {
-    const {disabled, value} = this.getThemeProps();
+    const {disabled, value} = this.props;
```

Comment:
I changed every `this.getThemeProps()` to `this.props`, is that ok? Is there another way to handle the error on `this.getThemeProps()`?

Reply:
That is correct, since the component is wrapped with `asBaseComponent` this is how it should be.